src/Mod/Spreadsheet/App/Sheet.cpp

```diff
@@ -867,6 +867,20 @@ void Sheet::getPropertyNamedList(std::vector<std::pair<const char*, Property*>>&
     }
 }
 
+bool Sheet::visitProperties(std::function<bool(App::Property*)> visitor) const
+{
+    if (!DocumentObject::visitProperties(visitor)) {
+        return false;
+    }
+    for (auto& v : cells.aliasProp) {
+        auto prop = getProperty(v.first);
+        if (prop && !visitor(prop)) {
+            return false;
+        }
+    }
+    return true;

```

Comment:
I hadn't considered the ordering semantics of the visit... methods I had added, though in my mind I envisioned them visiting the contents in the `begin` to `end` order and stopping when the lambda returns false.
`std::all_of` does not seem to define its visiting order at all, nor whether a return of false stops any further visits.
I'm not sure if it matters anywhere I did this, but perhaps the visit method should have a name that better reflects this.

For most collections, it would be sufficient to expose a begin and end method (and a perhaps return type to allow declaring of local variables), but the (current) organization of object Properties precludes this, so instead this visit... model was used

Reply:
You're raising a valid question. Though it doesn't seem to be explicitly stated by the standard I think it is implemented like that when the order is known, I only have this post to back up my statement: https://stackoverflow.com/a/47692337 I guess it could visit an extra element in parallel on a super optimized implementation.

Reply:
We can keep your implementation here if that exact order and sequential access is required. It's not a dealbreaker :)
Maybe a better name indicating the behavior as you suggested would be good though.

Reply:
Actually, I just reviewed it and in this case, not only does the order of visiting matter, the only outside call to this nest of related methods passes a lambda that always returns true, so all the properties always are visited.

As a result I'm changing the semantics of this method to not have any short-circuit return, and use std::for_each when appropriate. Now that I write this comment, it occurs to me that just going back to the `for (... : collection)` syntax would be simpler to read and does not require a lambda.